localnet: Make sure bridge mapping is valid

oshoval opened the pull request:
For localnet, the ovs `external_ids:ovn-bridge-mappings` should have entry for the network 
(either name, or physicalNetworkName if used). 
Validate it.

<!--
Please make sure you've read and understood our contributing guidelines;
https://github.com/ovn-kubernetes/ovn-kubernetes/blob/master/CONTRIBUTING.md

** Make sure all your commits include a signature generated with `git commit -s` **

All changes must adhere to this template to make it easy for reviewers
and preserve rationale/history behind every change
-->

## 📑 Description
<!-- Add a brief description of the pr -->

<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes #

## Additional Information for reviewers
<!--
What exactly did you change - you may also defer to information
contained in commit messages. At a bare minimum it's worth highlighting
which areas of the code were changed as it's easier to assign reviewers
-->

## ✅ Checks
<!-- Make sure your pr passes the CI checks and do check the following fields as needed - -->
- [ ] My code requires changes to the documentation
- [ ] if so, I have updated the documentation as required
- [x] My code requires tests
- [x] if so, I have added and/or updated the tests as required
- [ ] All the tests have passed in the CI <!-- If not leave a comment as to why the CI is red and if you need help understanding what's wrong -->

## How to verify it
<!--
Did you include unit tests? or end-to-end tests?
How can I manually verify that this patch achieves its objective
-->

maiqueb commented on go-controller/pkg/cni/types.go:
```diff
@@ -166,6 +166,9 @@ type PodRequest struct {
 
 	// the DeviceInfo struct
 	deviceInfo nadapi.DeviceInfo
+
+	// the topology type
+	toplogy string
```
drop this, it is not needed.

maiqueb commented on go-controller/pkg/cni/udn/primary_network.go:
```diff
@@ -25,6 +25,10 @@ func NewPrimaryNetwork(networkManager networkmanager.Interface) *UserDefinedPrim
 	}
 }
 
+func (p *UserDefinedPrimaryNetwork) TopologyType() string {
+	return p.activeNetwork.TopologyType()
+}
+
```
drop this, it is not needed.

maiqueb commented on go-controller/pkg/cni/cni.go:
```diff
@@ -345,6 +349,10 @@ func HandlePodRequest(
 // instance of the pod in the apiserver, see checkCancelSandbox for more info.
 // If kube api is not available from the CNI, pass nil to skip this check.
 func getCNIResult(pr *PodRequest, getter PodInfoGetter, podInterfaceInfo *PodInterfaceInfo) (*current.Result, error) {
+	if err := checkBridgeMapping(pr.toplogy, pr.netName); err != nil {
```
You already have the topology defined in the CNI conf. Did you try using it and it didn't work ? 

```suggestion
	if err := checkBridgeMapping(pr.CNIConf.Topology, pr.netName); err != nil {
```

maiqueb reviewed (changes requested):
Please check this version out. 

I think it should just work.

You're returning an error if the mapping is not found, which I think will crash-loop the pod, as we discussed with @jcaamano previously. 

Please provide some data on what happens when you schedule a pod and hte mapping is not available - do we see a proper error message when we describe the pod ?

maiqueb commented:
IMHO you'll also need an e2e test to assert this behavior - at the very least, add a new localnet describe block to https://github.com/ovn-kubernetes/ovn-kubernetes/blob/a3570ad9188f21b41b5ed2968d546a3161eaffb4/go-controller/pkg/cni/cni_test.go#L1 with some unit tests that exercise this - i.e. expect this error when the mapping is not found; expect nil when the mapping is found.

maiqueb commented:
> Please check this version out.
> 
> I think it should just work.
> 
> You're returning an error if the mapping is not found, which I think will crash-loop the pod, as we discussed with @jcaamano previously.
> 
> Please provide some data on what happens when you schedule a pod and hte mapping is not available - do we see a proper error message when we describe the pod ?

Just deployed the changes I proposed on top of yours, and the pod gets stuck in `ContainerCreating` and the following error is shown on a pod describe:
```
  Normal   AddedInterface          10s  multus   Add eth0 [10.244.0.3/24] from ovn-kubernetes
  Warning  FailedCreatePodSandBox  9s   kubelet  Failed to create pod sandbox: rpc error: code = Unknown desc = failed to setup network for sandbox "a482587c210a6a6145534d4e5c45be4758accda356c7eb58c9f776d1e437941d": plugin type="multus" name="multus-cni-network" failed (add): [default/tinypod/575876eb-a636-4c88-846e-fe45d2954fb7:tenantblack]: error adding container to network "tenantblack": CNI request failed with status 400: '[default/tinypod a482587c210a6a6145534d4e5c45be4758accda356c7eb58c9f776d1e437941d network tenantblack NAD default/localnet-network] [default/tinypod a482587c210a6a6145534d4e5c45be4758accda356c7eb58c9f776d1e437941d network tenantblack NAD default/localnet-network] failed bridge mapping validation: failed to find bridge mapping for network: "tenantblack"; Current ovn-bridge-mappings: "physnet:breth0"
```

oshoval commented:
> > Please check this version out.
> > I think it should just work.
> > You're returning an error if the mapping is not found, which I think will crash-loop the pod, as we discussed with @jcaamano previously.
> > Please provide some data on what happens when you schedule a pod and hte mapping is not available - do we see a proper error message when we describe the pod ?
> 
> Just deployed the changes I proposed on top of yours, and the pod gets stuck in `ContainerCreating` and the following error is shown on a pod describe:
> 
> ```
>   Normal   AddedInterface          10s  multus   Add eth0 [10.244.0.3/24] from ovn-kubernetes
>   Warning  FailedCreatePodSandBox  9s   kubelet  Failed to create pod sandbox: rpc error: code = Unknown desc = failed to setup network for sandbox "a482587c210a6a6145534d4e5c45be4758accda356c7eb58c9f776d1e437941d": plugin type="multus" name="multus-cni-network" failed (add): [default/tinypod/575876eb-a636-4c88-846e-fe45d2954fb7:tenantblack]: error adding container to network "tenantblack": CNI request failed with status 400: '[default/tinypod a482587c210a6a6145534d4e5c45be4758accda356c7eb58c9f776d1e437941d network tenantblack NAD default/localnet-network] [default/tinypod a482587c210a6a6145534d4e5c45be4758accda356c7eb58c9f776d1e437941d network tenantblack NAD default/localnet-network] failed bridge mapping validation: failed to find bridge mapping for network: "tenantblack"; Current ovn-bridge-mappings: "physnet:breth0"
> ```

good enough then right ?
thanks

maiqueb commented:
> > > Please check this version out.
> > > I think it should just work.
> > > You're returning an error if the mapping is not found, which I think will crash-loop the pod, as we discussed with @jcaamano previously.
> > > Please provide some data on what happens when you schedule a pod and hte mapping is not available - do we see a proper error message when we describe the pod ?
> > 
> > 
> > Just deployed the changes I proposed on top of yours, and the pod gets stuck in `ContainerCreating` and the following error is shown on a pod describe:
> > ```
> >   Normal   AddedInterface          10s  multus   Add eth0 [10.244.0.3/24] from ovn-kubernetes
> >   Warning  FailedCreatePodSandBox  9s   kubelet  Failed to create pod sandbox: rpc error: code = Unknown desc = failed to setup network for sandbox "a482587c210a6a6145534d4e5c45be4758accda356c7eb58c9f776d1e437941d": plugin type="multus" name="multus-cni-network" failed (add): [default/tinypod/575876eb-a636-4c88-846e-fe45d2954fb7:tenantblack]: error adding container to network "tenantblack": CNI request failed with status 400: '[default/tinypod a482587c210a6a6145534d4e5c45be4758accda356c7eb58c9f776d1e437941d network tenantblack NAD default/localnet-network] [default/tinypod a482587c210a6a6145534d4e5c45be4758accda356c7eb58c9f776d1e437941d network tenantblack NAD default/localnet-network] failed bridge mapping validation: failed to find bridge mapping for network: "tenantblack"; Current ovn-bridge-mappings: "physnet:breth0"
> > ```
> 
> good enough then right ? thanks

And the pod went into `Running` once I've manually added that mapping to the node where the pod was scheduled on.

IMHO yes.

Please focus on the tests.

oshoval commented:
Thanks, addressed comments / added UT
they are enough imho
testing directly `checkBridgeMapping` cases, and not the caller of that
because the use is simple (it just calls it) and even the current UT use stub for getCNIResult (`getCNIResultStub`)
so it isn't straight forward to test it, and also not that important

maiqueb commented:
> Thanks, addressed comments / added UT they are enough imho testing directly `checkBridgeMapping` cases, and not the caller of that because the use is simple (it just calls it) and even the current UT use stub for getCNIResult (`getCNIResultStub`) so it isn't straight forward to test it, and also not that important

But why do you need to invoke `checkBridgeMapping` from `getCNIResult` ? Can't you invoke that from the caller, which would enable you to test the integration of this function in the CNI flow ? 

What you'll run into trouble with then if you need to somehow mock the OVS DB - since you have that available in the unit tests.

oshoval commented:
The caller is `CmdAdd`, and since we can't change its prototype, it doesn't make sense to move it there
(also imho too "high" level there), moreover it will need mock indeed, and it is better to use [harness](https://github.com/ovn-kubernetes/ovn-kubernetes/pull/5145/files#diff-d38f452089bd09509003f301b9158ea207fb6865229d13e22d769357dff391bfR389) as done now (and on other tests),
the current call is very simple, so on those cases, it doesnt make sense to over engineer
we just create the ovs client and call the function, returning err in case of an error

oshoval commented:
fixed gci lint problems

oshoval commented:
should we address https://github.com/ovn-kubernetes/ovn-kubernetes/pull/5000/#discussion_r2000962389 in this PR as well ?
for that will need somehow to export the ovsClient, so few files can use it

oshoval commented:
WIP - need to fix the e2e tests as done on previous PR
https://github.com/ovn-kubernetes/ovn-kubernetes/actions/runs/14155936643/job/39655738416?pr=5145

oshoval commented:
rebased 
fixed e2e tests according previous PR - lets see it pass

oshoval commented:
need to see this one is not real problem
https://github.com/ovn-kubernetes/ovn-kubernetes/actions/runs/14156782786/job/39657620377?pr=5145
(wasnt on the previous PR)

oshoval commented:
```
Summarizing 1 Failure:
  [FAIL] Multi Homing multiple pods connected to the same OVN-K secondary network localnet OVN-K secondary network with a trunked configuration [BeforeEach] the same bridge mapping can be shared by a separate VLAN by using the physical network name attribute
  /root/go/src/github.com/ovn-kubernetes/test/e2e/multihoming.go:1261
```
seems it fails locally, even on clean main branch

trying to check why it has the bridge already created even that i run one test only,
wondering if we will see it on CI on other PRs as well
```
  STEP: setting up the localnet underlay with a trunked configuration @ 03/31/25 13:17:37.679
  [FAILED] in [BeforeEach] - /root/go/src/github.com/ovn-kubernetes/test/e2e/multihoming.go:1267 @ 03/31/25 13:17:37.832
  [PANICKED] in [AfterEach] - /root/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.22.0.linux-amd64/src/runtime/panic.go:261 @ 03/31/25 13:17:40.042

...

  [BeforeEach] /root/go/src/github.com/ovn-kubernetes/test/e2e/multihoming.go:1251
  [It] /root/go/src/github.com/ovn-kubernetes/test/e2e/multihoming.go:1295

  [FAILED] configuring the OVS bridge
  Expected success, but got an error:
      <*errors.errorString | 0xc0018149b0>: 
      failed to ADD OVS bridge ovsbr1: failed to run "kubectl -n ovn-kubernetes exec ovs-node-9s45z -- ovs-vsctl add-br ovsbr1": exit status 1 (ovs-vsctl: cannot create a bridge named ovsbr1 because a bridge named ovsbr1 already exists
      command terminated with exit code 1
      )
      {
          s: "failed to ADD OVS bridge ovsbr1: failed to run \"kubectl -n ovn-kubernetes exec ovs-node-9s45z -- ovs-vsctl add-br ovsbr1\": exit status 1 (ovs-vsctl: cannot create a bridge named ovsbr1 because a bridge named ovsbr1 already exists\ncommand terminated with exit code 1\n)",
      }
```

oshoval commented:
if using `export FLAKE_ATTEMPTS=0`
we can see this hidden problem
```
  [FAILED] in [BeforeEach] - /root/go/src/github.com/ovn-kubernetes/test/e2e/multihoming.go:1261 @ 03/31/25 14:46:48.677
  I0331 14:46:48.729550 3783263 util.go:1047] Unexpected error: 
      <*errors.errorString | 0xc000d4e070>: 
      failed to run "docker exec ovn-control-plane mkdir -p /var/log/e2e-dbs/the_same_bridge_mapping_can_be_shared_by_a_separate_VLAN_by_using_the_physical_network_name_attribute-multi-homing-6439": exit status 1 (Error response from daemon: client version 1.44 is too new. Maximum supported API version is 1.41
      )
      {
          s: "failed to run \"docker exec ovn-control-plane mkdir -p /var/log/e2e-dbs/the_same_bridge_mapping_can_be_shared_by_a_separate_VLAN_by_using_the_physical_network_name_attribute-multi-homing-6439\": exit status 1 (Error response from daemon: client version 1.44 is too new. Maximum supported API version is 1.41\n)",
      }
  [FAILED] in [JustAfterEach] - /root/go/src/github.com/ovn-kubernetes/test/e2e/util.go:1047 @ 03/31/25 14:46:48.729
  [PANICKED] in [AfterEach] - /root/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.22.0.linux-amd64/src/runtime/panic.go:261 @ 03/31/25 14:46:48.729
  STEP: dump namespace information after failure @ 03/31/25 14:46:48.73
```

so using the following, before cluster deployment, fixes the problem, maybe the github CI bootstrap was updated behind the scene?
```
export DOCKER_API_VERSION=1.43
sudo systemctl restart docker
```

qinqon commented:
> interesting point when doing it from CNI (that didnt happen on the previous PR design) is that the prefix is important i.e
> 
> ```
> ... failed to find bridge mapping for network: "q2cl9_different-network"; Current ovn-bridge-mappings: "physnet:breth0,different-network:ovsbr1"
> ```
> 
> there is a random internal prefix that is added somewhere in the flow, in this example `q2cl9_`
> 
> There are 2 options:
> 
> 1. in the failing test i can either read the NAD and add according the actual network name create the mapping
> 2. to remove the prefix in CNI if exists (i.e remove all before `_`, including it)
> 
> wdyt?

First you need to understand what this prefix is.

oshoval commented:
@qinqon can you please review ?
need to see all test pass, but meanwhile if you want we can start the review
thanks

oshoval commented on go-controller/pkg/cni/cni.go:
```diff
@@ -345,6 +350,23 @@ func HandlePodRequest(
 // instance of the pod in the apiserver, see checkCancelSandbox for more info.
 // If kube api is not available from the CNI, pass nil to skip this check.
 func getCNIResult(pr *PodRequest, getter PodInfoGetter, podInterfaceInfo *PodInterfaceInfo) (*current.Result, error) {
+	ctx := context.Background()
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	ovsClient, err := libovsdb.NewOVSClient(ctx.Done())
+	if err != nil {
+		return nil, fmt.Errorf("failed to create ovs client: %w", err)
+	}
+
+	netName := pr.netName
+	if pr.CNIConf.PhysicalNetworkName != "" {
+		netName = pr.CNIConf.PhysicalNetworkName
+	}
+	if err := checkBridgeMapping(ovsClient, pr.CNIConf.Topology, netName); err != nil {
+		return nil, fmt.Errorf("failed bridge mapping validation: %w", err)
+	}
+
```
we already discussed with Miguel and saw the `NewOVSClient` is the preferred way, as it has cache
there was a PR that adapted from the old method to the new method as far as i remember
also please note that in the previous review, Jaime was not against it, at most asked to reuse the ovsClient that was already created on other place 

but it means we will need to create a singleton fetcher for that, if they are in the same container, do we want this ?
https://github.com/ovn-kubernetes/ovn-kubernetes/pull/5000/#discussion_r2000962389

qinqon replied:
at least can we move those ovs operations to ovs.go ?

oshoval replied:
i think it will mix the methods which isnt good
it can be confusing, even now we dont have them mixed where `NewOVSClient` is used

qinqon replied:
To be honest I would just make this PR simple to fix the issue at hand and do like the other ovs operations, and if you want after merging it refactor all the ovs operations with ovs client, but whatever maintainers prefer.

oshoval replied:
we already had agreement that this is the preferred method, i dont think we need to go back please

oshoval replied:
> UPDATE: at privileged mode we use the CNI server that is running at ovnkube-node controllers

does it mean we should strive to create singleton then ?
what about unprivileged mode, if one mode works and the other isnt, it means that we either forget about it, or simply the singleton will run in two places, each one of them will init it in the worse case, which is fine

qinqon replied:
It means that we may need to detect the mode and re-use the controller cache if we don't do that we end with a pair of ovs caches at privileged mode.

oshoval replied:
what about having `once.Do(func() {`
and the rest is automatic, i.e:
if they both run from the same container, it will simply reuse the already created instance on follow-up calls,
otherwise each first call will init it, on different containers, without no need to detect the mode

oshoval replied:
np thanks, once Jaime will review we can see if still needed or not

oshoval replied:
@jcaamano @qinqon @maiqueb
wrt Jaime's comment
https://github.com/ovn-kubernetes/ovn-kubernetes/pull/5000#discussion_r2000962389

Do we still want to do it on this PR ?
considering what Quique said, that there is privileged and unprivileged mode,
we can create a struct with `once.Do`  (sync pkg), that initialize the client on the first run, and return it on subsequent ones.
It can be agnostic to the mode iiuc, because if it runs on two containers, each one will simply init the client for its own, otherwise the 2nd call onwards will just use the singleton

Thanks

maiqueb replied:
I think we should do it now @oshoval

oshoval commented on go-controller/pkg/cni/cni.go:
```diff
@@ -420,3 +442,26 @@ func (pr *PodRequest) buildPodInterfaceInfo(annotations map[string]string, podAn
+		return nil
+	}
+
+	openvSwitch, err := ovsops.GetOpenvSwitch(ovsClient)
+	if err != nil {
+		return fmt.Errorf("failed getting openvswitch: %w", err)
+	}
+
+	ovnBridgeMappings := openvSwitch.ExternalIDs["ovn-bridge-mappings"]
+
+	bridgeMappings := strings.Split(ovnBridgeMappings, ",")
+	for _, bridgeMapping := range bridgeMappings {
+		networkBridgeAssociation := strings.Split(bridgeMapping, ":")
+		if len(networkBridgeAssociation) == 2 && networkBridgeAssociation[0] == networkName {
+			return nil
+		}
+	}
```
we dont need to use ovs.go please see https://github.com/ovn-kubernetes/ovn-kubernetes/pull/5145#discussion_r2022704248

unless you insist, i dont think we need intermediate conversion

qinqon replied:
ack

qinqon reviewed (dismissed):
LGTM

We have alreay review 99% of this.

oshoval commented:
all tests passed, ptal (just rebased and added the 2nd commit to address [this](https://github.com/ovn-kubernetes/ovn-kubernetes/pull/5145#discussion_r2031320764))

maiqueb reviewed (approved):
thank you. 

@jcaamano / @ricky-rav are you OK with these changes ?

oshoval commented:
On unpriviledge mode, there can be a subtle bug, that if the first time failed to init the client, we will reach a limbo state, where the pod is alive, but it always fail to get the ovsClient, because we use sync once.
Fixed it to allow reinit attempts in case of a failure

oshoval commented:
@maiqueb ptal due to the latest changes

ricky-rav reviewed (dismissed):
/LGTM
Thanks, Or!

jcaamano commented on go-controller/cmd/ovnkube/ovnkube.go:
```diff
@@ -561,16 +562,17 @@ func runOvnKube(ctx context.Context, runMode *ovnkubeRunMode, ovnClientset *util
 		}()
 	}
 
+	ovsClient, err := ovs.GetOVSClient(ctx)
```
Prefer if you pass this just like we do with everything else.

NewNodeControllerManager -> NewDefaultNodeNetworkController -> NewCNIServer. And form here you just pass ti around same as the clientset.

I don't have anything against singletons, but I would prefer not start popping them without some kind of central registry design or something. Otherwise its easy to lose track of them.

I think passing it around is the best approach for now.

jcaamano replied:
Also I don't think you need to change anything relative to `cancel()`. Just leave as it was. If anything, you can remove the call there was online 568 which should be completely unnecessary.

jcaamano commented:
Also. Are the IC only support references in the commit & PR message accurate? I don't see why this wouldn't work in non-IC.

oshoval commented:
> Also. Are the IC only support references in the commit & PR message accurate? I don't see why this wouldn't work in non-IC.

interesting indeed, since now it is on CNI level, CNI is per node anyhow which is the thing we need

oshoval commented:
> Also. Are the IC only support references in the commit & PR message accurate? I don't see why this wouldn't work in non-IC.

Interesting, indeed, checking on non IC works, since it is from the CNI now
is it correct that on CNI it always runs on the specific node ? (i guess because it should enter its network ns, but best to ask if it is the case on all the sub configurations)
will update code / commits accordingly

oshoval commented:
addressed comments, lets see tests are passing
non IC is supported now, nice catch
thanks

oshoval commented:
all passed beside known flakes
changes as Jaime asked - support non IC (just removed the condition) and pass the client via functions instead of singleton

oshoval commented:
@jcaamano @maiqueb @qinqon can you please take a look ?
Thanks

maiqueb commented on test/e2e/multihoming.go:
shouldn't we add a test entry that triggers the failure ? I'm somehow missing that.

oshoval replied:
imo we don't need to test both bad and good path
we test the good path e2e, and the unit test tests the bad path

maiqueb replied:
I think it would be a good test to add, but, at least it's covered in unit tests. 

Up to the maintainer to say if this is really needed.

jcaamano commented on go-controller/cmd/ovnkube/ovnkube.go:
```diff
@@ -521,6 +521,12 @@ func runOvnKube(ctx context.Context, runMode *ovnkubeRunMode, ovnClientset *util
 		}()
 	}
 
+	ovsClient, err := libovsdb.NewOVSClient(ctx.Done())
+	if err != nil {
+		cancel()
+		return fmt.Errorf("failed to initialize libovsdb vswitchd client: %w", err)
+	}
+
```
This should only be created when `runMode.node`

The usage we had below for metrics seems to only be IC compatible.

I would either return an error if metrics have been requested on non-node configuration or alternatively not start the metric server or not register the metrics in that case.

But it seems a somewhat incorrect metrics design. @martinkennelly is there anything I am missing here?

martinkennelly replied:
whats metrics are you referring to that are ic only?

ormergi commented:
[Changes](https://github.com/ovn-kubernetes/ovn-kubernetes/compare/27c795586501edd4c8627e0e9cb8ec721fa2343b..2fe493dd1b64bb547fc71cf824f40631095bad67):
- Init. OVS client when only when necessary, resolve https://github.com/ovn-kubernetes/ovn-kubernetes/pull/5145#pullrequestreview-2787344038 
- In the CNI bridge-mapping check, remove existing node bridge-mapping from the error message in order to avoid exposing this info to non-admin users, as the error is propagated to pod event from kubelet any user can have access to it (using kubectl describe pod..)

@jcaamano @martinkennelly PTAL

ormergi commented:
[Changes](https://github.com/ovn-kubernetes/ovn-kubernetes/compare/27c795586501edd4c8627e0e9cb8ec721fa2343b..6878d61eec42ccbeb0785149ff0fe6e35af70e89):
- Initialize OVS client when needed, resolve https://github.com/ovn-kubernetes/ovn-kubernetes/pull/5145#pullrequestreview-2787344038
- Avoid exposing the existing nodes OVN bridge-mapping in CNI error message, as the error eventually propogated as pod event any user can access (using kubectl describe pod ..)

jcaamano commented on go-controller/pkg/cni/cni.go:
```diff
@@ -420,3 +435,26 @@ func (pr *PodRequest) buildPodInterfaceInfo(annotations map[string]string, podAn
+			return nil
+		}
+	}
+	klog.Errorf("Failed to find bridge mapping for network: %q, current OVN bridge-mappings: (%s)", networkName, ovnBridgeMappings)
```
nit: logging should happen top level. Difficult for me to tell if this is not being logged twice.

ormergi replied:
Unfortunately the existing bridge mapping info exist in this context, this is why I added the log message here.
I checked how logs will look in case of an error, ovnkube-contoller logs will have the following messages:
```
E0424 15:49:43.805811  117892 cni.go:458] Failed to find bridge mapping for network: "mylocalnet", current OVN bridge-mappings: (physnet:breth0)
I0424 15:49:43.805823  117892 cni.go:348] [default/test 9acfbd9f1e5a4e5c6ada7e352cb43fc16ec03c8c3bbcf9c46894bfffddcdd074 network localnet NAD default/localnet] ADD finished CNI request [default/test 9acfbd9f1e5a4e5c6ada7e352cb43fc16ec03c8c3bbcf9c46894bfffddcdd074 network localnet NAD default/localnet], result "", err failed bridge mapping validation: failed to find OVN bridge-mapping for network: "mylocalnet"
```
First one the log message in this scope.
The second message comes from the CNI top-level log message, which doesnt include the node info.

jcaamano replied:
I suggest to include the existing bridge mappings on the error or if that is too verbose then keep the log but change it to v(5)?

ormergi replied:
I am trying to prevent having the nodes bridge-mapping in the error message, because the emitted pod event message derives from that error including the node info, which is accesiable for non-admin users.

I changed the log level to v(5) as you suggest.

ormergi commented:
[Changes](https://github.com/ovn-kubernetes/ovn-kubernetes/compare/6878d61eec42ccbeb0785149ff0fe6e35af70e89..5ac9a14bec3be41edb8b7308302f581c7193e7bc): resolve https://github.com/ovn-kubernetes/ovn-kubernetes/pull/5145#discussion_r2058714603
@jcaamano PTAL :pray:
